native/components/message-markdown-rules.react.js

```diff
@@ -0,0 +1,136 @@
+      ...SimpleMarkdown.defaultRules.autolink,
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => {
+        state.withinText = true;
+        const _pressHandler = () => {
+          pressHandler(node.target);
+        };
+        const autolink = (
+          <Text key={state.key} style={styles.autolink} onPress={_pressHandler}>
+            {output(node.content, state)}
+          </Text>
+        );
+        return autolink;
+      },
+    },
+    link: {
+      ...SimpleMarkdown.defaultRules.link,
+      match: SimpleMarkdown.inlineRegex(
+        new RegExp(
+          '^\\[(' + LINK_INSIDE + ')\\]\\(' + LINK_HREF_AND_TITLE + '\\)',
+        ),
+      ),
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => {
+        state.withinLink = true;
+        const _pressHandler = () => {
+          pressHandler(node.target);
+        };
+        const link = (
+          <Text key={state.key} style={styles.autolink} onPress={_pressHandler}>
+            {output(node.content, state)}
+          </Text>
+        );
+        state.withinLink = false;
+        return link;
+      },
+    },
+    // in some cases, simple-markdown recognizes "normal" text as a paragraph
+    // and we need this rule to preserve our style and bubble size
+    paragraph: {
+      ...SimpleMarkdown.defaultRules.paragraph,
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => {
+        const paragraph = (
+          <Text key={state.key} style={styles.paragraph}>
+            {output(node.content, state)}
+          </Text>
+        );
+        return paragraph;
+      },
+    },
+    // rule to preserve our text style in simple messages
+    text: {
+      ...SimpleMarkdown.defaultRules.text,
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<string>,
+        state: SimpleMarkdown.State,
+      ) => {
+        let textStyle = {
+          ...styles.text,
+          ...(state.style || {}),
+        };
+
+        if (state.withinLink) {
+          textStyle = [styles.text, styles.autolink];
+        }
+
+        const text = (
+          <Text key={state.key} style={textStyle}>
+            {node.content}
+          </Text>
+        );
+        return text;
+      },
+    },
+    //simple-markdown uses url rule to catch https inside text
+    url: {
+      ...SimpleMarkdown.defaultRules.url,
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => {
+        state.withinText = true;
+        const _pressHandler = () => {
```

Comment:
I am concerned that so much of this code still seems to be copy-pasted. We talked previously about how you shouldn't copy-paste code from other packages. You can look it at side-by-side, and use it as inspiration. But when you name the variables the same and define the functions in the same way, it seems that you are copy-pasting.

This is why I suggested in my previous review that you start over from scratch. The fact that you started this PR by copy-pasting code is an issue. If you really don't want to start over, you need to make sure that you go through all of this code and rewrite every line.